src/test/java/net/revelc/code/formatter/AbstractFormatterTest.java

```diff
@@ -73,16 +73,19 @@ public String getCompilerCodegenTargetPlatform() {
         }
     }
 
-    protected void doTestFormat(Formatter formatter, String fileUnderTest, String expectedSha512) throws IOException {
+    protected void doTestFormat(Map<String, String> options, Formatter formatter, String fileUnderTest,
+            String expectedSha512) throws IOException {
+
         File originalSourceFile = new File("src/test/resources/", fileUnderTest);
         File sourceFile = new File("target/testoutput/", fileUnderTest);
+        Files.copy(originalSourceFile, sourceFile);
```

Comment:
It's not clear to me why this file copy was moved to above the code that created its parent directory. If the copy handles the case where the directory doesn't exist, then those two lines for creating `targetDir` can be removed. Otherwise, they need to precede the copy.

Reply:
I just added an overloaded method that accepts additional param `options`.  These `options` when set formtter.init(options,...) where getting overwritten with a empty `options` in the original method. In the overloaded method, I check if its not set already and then init it to empty `options`

Reply:
I think you missed my point. The destination in this copy either 1) does not exist yet, because the `mkdir` line is still below this, or 2) will get created, in which case the below `mkdir` line is redundant. Either way, a "copy to directory X" statement should not precede a "create directory X" statement.